java-wasm/src/main/java/org/prism/Prism.java

```diff
@@ -22,7 +22,7 @@ public Prism() {
     public Prism(WasiOptions wasiOpts) {
         wasi = WasiPreview1.builder().withOptions(wasiOpts).build();
         instance = Instance.builder(PrismModule.load())
-            .withMemoryFactory(ByteArrayMemory::new)
+            .withMemoryFactory(limits -> new ByteArrayMemory(new MemoryLimits(10, MemoryLimits.MAX_PAGES)))
```

Comment:
- nitpick - it will be good if the memory limits can be tuned when creating the object.

Maybe in the next iteration we can make use of a `Builder` to lock the constructor down.